src/views/ModelPlan/TaskList/Beneficiaries/Frequency/index.tsx

```diff
@@ -362,28 +369,75 @@ const Frequency = () => {
                         scrollElement="precedenceRules"
                         error={!!flatErrors.precedenceRules}
                       >
-                        <Label
-                          htmlFor="beneficiaries-precedence-rules"
-                          className="maxw-none"
-                        >
-                          {beneficiariesT('precedenceRules.label')}
-                        </Label>
-
-                        <p className="text-base margin-0 line-height-body-3">
-                          {beneficiariesT('precedenceRules.sublabel')}
-                        </p>
-
-                        <FieldErrorMsg>
-                          {flatErrors.precedenceRules}
-                        </FieldErrorMsg>
-
-                        <Field
-                          as={TextAreaField}
-                          className="height-15"
-                          error={flatErrors.precedenceRules}
-                          id="beneficiaries-precedence-rules"
-                          data-testid="beneficiaries-precedence-rules"
+                        <FieldArray
                           name="precedenceRules"
+                          render={arrayHelpers => (
+                            <>
+                              <legend className="usa-label">
+                                {beneficiariesT('precedenceRules.label')}
+                              </legend>
+
+                              <p className="text-base margin-top-1 margin-bottom-0 line-height-body-3">
+                                {beneficiariesT('precedenceRules.sublabel')}
+                              </p>
+
+                              <FieldErrorMsg>
+                                {flatErrors.precedenceRules}
+                              </FieldErrorMsg>
+
+                              {getKeys(beneficiaryPrecedenceConfig.options).map(
+                                key => (
+                                  <Fragment key={key}>
+                                    <Field
+                                      as={CheckboxField}
+                                      id={`beneficiaries-precedence-rules-${key}`}
+                                      data-testid={`beneficiaries-precedence-rules-${key}`}
+                                      name={`precedence-rules-${key}`}
```

Comment:
The value `name` here is the most important part of the Formik component.  It isn't merely a name, but the actually fileld name where the value is tied to the Formik library.  It should be `precedenceRules` that houses both values for yes and no